### PR TITLE
ISPN-5033 Allow undefining configurations

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-5033

### CODE REVIEW

**Comment by wburns** [comment]
Seems we should throw an exception if a running cache is using this configuration?  Also if we have a stopped cache for this name seems we should remove it?

**Comment by wburns** [comment]
Although now I see that the cache configuration stop does the removal as well.  In that case do we even have to make the undefining of cache configuration public?

**Comment by tristantarrant** [comment]
@wburns I've added a check to see if a configuration is in use. I prefer to have it in as a public API for symmetry with defineConfiguration()

File: core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java [hunk 1]
```diff
@@ -88,6 +88,14 @@
    Configuration defineConfiguration(String cacheName, String templateCacheName, Configuration configurationOverride);
 
    /**
+    * Removes a configuration from the set of defined configurations. If the named configuration does not exist, nothing
+    * happens
+    *
+    * @param configurationName     the named configuration
+    */
+   void undefineConfiguration(String configurationName);
```
Comment (wburns) [inline core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java]: We should add a note in the Javadoc here about the IllegalStateException, we could even add it as throws in the method declaration.

**Comment by tristantarrant** [comment]
I've extended the javadoc. Failures are unrelated to this change.

**Comment by wburns** [comment]
Pulling...

**Comment by wburns** [comment]
Integrated into master, thanks @tristantarrant !

**Pull request closed without merging**